lib/ecto/changeset.ex

```diff
@@ -2054,7 +2054,7 @@ defmodule Ecto.Changeset do
     constraint = opts[:name] || "#{get_source(changeset)}_#{get_field_source(changeset, field)}_index"
     message    = message(opts, "has already been taken")
     match_type = Keyword.get(opts, :match, :exact)
-    add_constraint(changeset, :unique, to_string(constraint), match_type, field, {message, []})
+    add_constraint(changeset, :unique, to_string(constraint), match_type, field, {message, [validation: :unique]})
```

Comment:
@josevalim I am trying to use `validation` key for represent the name for the validation, the reason is that I could do things like this https://github.com/straw-hat-llc/straw_hat/blob/5195fcaae9c1dbd2a71b645782d9e0096d351d27/lib/straw_hat/error/changeset_parser.ex#L33

Where  I will use it for Internationalization propose in this case, or for create some unique key.

I wouldn't mind adding more information but at least get the concern of keeping `validation` key for that. It was the whole root of my issue 😄